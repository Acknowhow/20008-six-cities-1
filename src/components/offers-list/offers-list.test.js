import React from 'react';
import renderer from 'react-test-renderer';
import OffersList from './offers-list.jsx';

const mock = {
  offers: [
    {
      mark: `Poor Quality`,
      name: `Dirty and cold`,
      src: `test.jpg`,
      type: `Hangover suite`,
      ratings: [
        1, 3, 4, 5, 5
      ],
      price: {
        period: `night`,
        value: 110
      }
    },
    {
      mark: `Premium`,
      name: `Wood and stone place`,
      src: `test1.jpg`,
      type: `Apartment`,
      ratings: [
        1, 2, 4, 2
      ],
      price: {
        period: `night`,
        value: 100
      }
    },
    {
      mark: `Poorest quality`,
      name: `Lo and behold`,
      src: `test2.jpg`,
      type: `Pigeon hole`,
      ratings: [
        2, 3, 4, 2, 4, 1, 5, 5
      ],
      price: {
        period: `night`,
        value: 90
      }
    },
    {
      mark: `Premium`,
      name: `Nice flat in the city center`,
      src: `img/apartment-01.jpg`,
      type: `Apartment`,
      ratings: [
        3, 5, 4, 3, 4, 5, 5, 5
      ],
      price: {
        period: `night`,
        value: 115
      }
    }
  ]
};

it(`App correctly renders after relaunch`, () => {

  const {offers} = mock;
  const tree = renderer
    .create(<OffersList
      offers={offers}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
