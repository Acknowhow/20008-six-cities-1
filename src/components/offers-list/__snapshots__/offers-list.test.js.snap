// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Poor Quality
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          data-index={0}
          height="200"
          onClick={[Function]}
          src="test.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            110
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "72%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__link"
          data-index={0}
          href="#"
          onMouseOver={[Function]}
        >
          Dirty and cold
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Hangover suite
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          data-index={1}
          height="200"
          onClick={[Function]}
          src="test1.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "45%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__link"
          data-index={1}
          href="#"
          onMouseOver={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Poorest quality
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          data-index={2}
          height="200"
          onClick={[Function]}
          src="test2.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            90
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "65%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__link"
          data-index={2}
          href="#"
          onMouseOver={[Function]}
        >
          Lo and behold
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Pigeon hole
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          data-index={3}
          height="200"
          onClick={[Function]}
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            115
          </b>
          <span
            className="place-card__price-text"
          >
            / 
            night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "85%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          className="place-card__link"
          data-index={3}
          href="#"
          onMouseOver={[Function]}
        >
          Nice flat in the city center
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartment
      </p>
    </div>
  </article>
</div>
`;
